packages/wonder-blocks-labeled-field/src/components/__tests__/labeled-field.test.tsx

```diff
@@ -215,11 +230,240 @@ describe("LabeledField", () => {
+            const error = "Error message";
+            render(
+                <LabeledField
+                    field={<TextField value="" onChange={() => {}} />}
+                    testId={testId}
+                    label={label}
+                    description={description}
+                    error={error}
+                />,
+                defaultOptions,
+            );
+
+            // Act
+            const labelEl = screen.getByText(label);
+            const descriptionEl = screen.getByText(description);
+            const fieldEl = screen.getByRole("textbox");
+            const errorEl = screen.getByText(error);
+
+            // Assert
+            expect(labelEl).toHaveAttribute("data-testid", `${testId}-label`);
+            expect(descriptionEl).toHaveAttribute(
+                "data-testid",
+                `${testId}-description`,
+            );
+            expect(fieldEl).toHaveAttribute("data-testid", `${testId}-field`);
+            expect(errorEl).toHaveAttribute("data-testid", `${testId}-error`);
+        });
+
+        it("should not include testId on elements if testId prop is not provided", () => {
+            // Arrange
+            const label = "Label";
+            const description = "Description of the field";
+            const error = "Error message";
+            render(
+                <LabeledField
+                    field={<TextField value="" onChange={() => {}} />}
+                    label={label}
+                    description={description}
+                    error={error}
+                />,
+                defaultOptions,
+            );
+
+            // Act
+            const labelEl = screen.getByText(label);
+            const descriptionEl = screen.getByText(description);
+            const fieldEl = screen.getByRole("textbox");
+            const errorEl = screen.getByText(error);
+
+            // Assert
+            expect(labelEl).not.toHaveAttribute("data-testid");
+            expect(descriptionEl).not.toHaveAttribute("data-testid");
+            expect(fieldEl).not.toHaveAttribute("data-testid");
+            expect(errorEl).not.toHaveAttribute("data-testid");
+        });
+
+        it("should persist original attributes on the field if it is not overridden", () => {
+            // Arrange
+            render(
+                <LabeledField
+                    field={
+                        <TextField
+                            value=""
+                            onChange={() => {}}
+                            name="name-example"
+                        />
+                    }
+                    label="Label"
+                />,
+                defaultOptions,
+            );
+
+            // Act
+            const fieldEl = screen.getByRole("textbox");
+
+            // Assert
+            expect(fieldEl).toHaveAttribute("name", "name-example");
+        });
+    });
+
+    describe("Accessibility", () => {
```

Comment:
**question:** Follow up question to one of my prev comments. Is there a test that verifies that the error is logged correctly by checking the element with `role="alert"`? I think this would be helpful in the future to prevent any possible regressions (disregard this comment if there's already a test verifying that).

Reply:
There are some tests checking for the alert! I will also make sure we have enough test coverage around errors when I re-visit error validation!

That said, I'm not sure yet if we should be using `role="alert"` for the error messages, depending on how we want to handle error validation. Since there are cases for submitting a form and getting multiple errors, would it be too much to have all the error messages as alerts? I will be bringing up error validation in our cubby meeting later to get some thoughts on this and we can discuss more!

Reply:
I think whether to use `role="alert"` depends if you're sending focus to the first invalid field. An alert announcement and a focused element announcement would compete with each other, so it's kind of one or the other.

Reply:
That makes sense! Thinking aloud, we may need to provide a prop that let's the devs decide the behaviour:
- If it's client side validation and the error message is shown onBlur, it would make sense for the error message to have `role="alert"` so it's announced right away
- If the error message is shown after form submission (usually from backend validation), we would focus on the first field with error message (and the error message doesn't have `role="alert"`

But what about when there's both client side validation and backend validation? 🤔 

I tested to see what happens in VoiceOver when multiple error messages with `role="alert"` are shown in the [validation example](https://5e1bf4b385e3fb0020b7073c-cmzmctcwqt.chromatic.com/?path=/story/packages-labeledfield--validation). It only announced the last error message! But maybe that's okay since a user would then fill out that field and get feedback again on a previous error?

I will look into this more when I work on the validation part, but adding some more thoughts here for now. Let me know if anyone has more ideas or alternatives!

Reply:
All of this makes sense, thanks for the context!

As an option, I was thinking that maybe it would be worth experimenting with setting `role="status"` by default (which is `polite` + `aria-atomic="true"`), and allowing to override to `alert` when needed.

Reply:
Good thinking and research here @beaesguerra! It's challenging that screen reader behavior is so inconsistent cross-platform. 

A few more thoughts/questions that come to mind after chatting about it:

- Exposing an `alert` client-side onBlur makes a ton of sense. For some reason [this example](https://www.w3.org/WAI/tutorials/forms/notifications/#on-focus-change) doesn't work in Safari and VO but does work in iOS Safari and VO.
    - Maybe this technique could be improved with the forthcoming Announcer component?
- A summary of error messages would be helpful in the event there are multiple errors. Ideally it would be visible, but at the very least something visually hidden could be announced to inform screen reader users. Would that cover the case of client-side validation and backend validation (were you meaning at the same time)? 
- Recipes and examples that we can test thoroughly will be helpful in deciding what to do! I'll be happy to help test when needed.

Reply:
Thanks for summarizing Marcy! We can definitely iterate on this to make it more consistent cross-platform :) 

> Would that cover the case of client-side validation and backend validation (were you meaning at the same time)

Yes! That would address both types of validation 👍 

> Recipes and examples that we can test thoroughly will be helpful in deciding what to do! I'll be happy to help test when needed.

I'll be working on some examples next and will appreciate the feedback!